### PR TITLE
Type guards

### PR DESCRIPTION
**Pull request by philipstanislaus**

### CODE REVIEW

File: components/ValueNFT/index.tsx [hunk 1]
```diff
@@ -87,17 +87,16 @@ const ValueNFT: React.FC<Props> = (props: Props) => {
   }
 
   const updateMaturityDate = async () => {
+    if (isTinlakeV2(props.tinlake)) throw Error('Tinlake V2 does not have maturity date')
```
Comment (offerijns) [inline components/ValueNFT/index.tsx:90]: Does this work? I always do `throw new Error()`. Isn't that a class you have to instantiate?

File: components/ValueNFT/index.tsx [hunk 2]
```diff
@@ -72,13 +73,12 @@ const ValueNFT: React.FC<Props> = (props: Props) => {
   const valueNFT = async () => {
     await props.ensureAuthed!()
 
-    const nftFeedId =
-      props.tinlake.version === 2
-        ? await (props.tinlake as ITinlake).getNftFeedId(registry, Number(tokenId))
-        : await (props.tinlake as ITinlakeV3).getNftFeedId(registry, tokenId)
+    const nftFeedId = isTinlakeV3(props.tinlake)
```
Comment (offerijns) [inline components/ValueNFT/index.tsx:76]: Much better 👍

**Review by offerijns — COMMENTED** [review]
Definitely a big improvement!

File: components/ValueNFT/index.tsx (hunk 1 above)
Comment (philipstanislaus) [reply components/ValueNFT/index.tsx:90]: Yes you are right. Weird that there's no type error.

Comment (philipstanislaus) [reply components/ValueNFT/index.tsx]: Fixed

**Pull request merged**